File: Coding_Platforms/Leetcode/Practice/Golang/strict_increase.go
package main

import "fmt"

func main() {

	fmt.Println(canBeIncreasing([]int{105, 924, 32, 968}))
}

func canBeIncreasing(nums []int) bool { //check increasing number

	var nums1 = make([]int, len(nums))

	if check_increase(nums) { // if increasing then return true
		return true

	} else {
		for i, _ := range nums { // range over slice 

			//copy slice into another slice and remove one number every time and check increase order
			if copy(nums1, nums);check_increase(append(nums1[:i], nums1[i+1:]...)) {
				return true
			}
		}
	}
	return false
}

func check_increase(num []int) bool { // check the increase order

	for idx, i := range num {//iterate over slice

		// if previous number is greater then current number then return false
		if (idx > 0) && (num[idx-1] >= i) { 
			return false
		}
	}
	return true
}

File: Coding_Platforms/Leetcode/Practice/Python/strict_increase.py
class Solution:
    def canBeIncreasing(self, nums: list[int]) -> bool:

        
        for i in range(len(nums)): # itrerate over length of nums

            new_nums = nums[:i]+nums[i+1:] # store nums in new_nums, of removing 1 element every time.
            if (new_nums == sorted(new_nums)) & (len(new_nums) == len(set(new_nums))): # if after removing 1 element all the list is in increase order then return true
                return True
            
        if (nums == sorted(nums)) & (len(nums) == len(set(nums))): # if all the list is in increase order then return true
            return True

        return False

print(Solution.canBeIncreasing(Solution,[1,1]))